### PR TITLE
[CASSH] Major security upgrade in 1.5.0

### PR DESCRIPTION
**Pull request by nbeguier**
1.5.0
-----

2018/08/09

### Changes

  - Every GET routes are DEPRECATED.
  - Authentication is in the payload now
  - Update tests

```
[OK] Test ping
[OK] Test health
[OK] Test status unknown user
[OK] Test deprecated status URL without username
[OK] Test deprecated admin status URL without username
[OK] Test deprecated status URL with username
[OK] Test deprecated status URL with username
[OK] Test add user without username
[OK] Test add user with bad username
[OK] Test add user without realname
[OK] Test add user with no pubkey
[OK] Test add user with bad pubkey
[OK] Test add user
[OK] Test add user named 'all' (should fail)
[OK] Test status pending user
[OK] Test add user with same realname (which is possible)
[OK] Test updating user
[OK] Test add user with same username (should fail)
[OK] Test signing key without username
[OK] Test signing key without realname
[OK] Test signing key with no pubkey
[OK] Test signing key with bad pubkey
[OK] Test signing key when wrong public key
[OK] Test signing key when PENDING status
[OK] Test admin revoke 'toto'
[OK] Test admin verify 'toto' status
[OK] Test signing key when revoked
[OK] Test delete 'toto'
[OK] Test admin active unknown user
[OK] Test admin verify 'testuser' status
[OK] Test admin active testuser
[OK] Test admin re-active testuser
[OK] Test signing key
[OK] Test delete 'testuser'
```

### CODE REVIEW

File: cassh [hunk 1]
```diff
@@ -110,44 +103,49 @@ class CASSH(object):
...
+            data.update(prefix)
+        return data
 
     def admin(self, username, action, set_value=None):
         """
         Admin CLI
         """
+        payload = self.get_data()
         try:
             if action == 'revoke':
```
Comment (vdemonchy) [inline cassh:124]: FAR (For Another Review): a switch/case would be more relevant instead of multiple if/elif conditions

**Review by vdemonchy — APPROVED** [review]
LGTM

File: cassh (hunk 1 above)
Comment (xakraz) [reply cassh:124]: `switch` does not exist in python ....

Comment (vdemonchy) [reply cassh:124]: You're goddamn right ! 
*facepalms himself*

**Pull request merged**